src/test/regress/upgrade/custom_citus.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+
+"""custom_citus
+Usage:
+    custom_citus --bindir=<bindir> --pgxsdir=<pgxsdir> --parallel=<parallel>
+
+Options:
+    --bindir=<bindir>              The PostgreSQL executable directory(ex: '~/.pgenv/pgsql-11.3/bin')
+    --pgxsdir=<pgxsdir>           	       Path to the PGXS directory(ex: ~/.pgenv/src/postgresql-11.3)
+    --parallel=<parallel>           how many configs to run in parallel
+"""
+
+import upgrade_common as common
+import threading
+import atexit
+from docopt import docopt
+import os, shutil
+import time
+import sys
+
+from config import (
+    CitusDefaultClusterConfig, CitusSingleNodeClusterConfig,
+    CitusSingleNodeSingleShardClusterConfig,
+    CitusSingleShardClusterConfig,
+    CitusNonMxClusterConfig,
+    CitusManyShardsClusterConfig,
+    CitusSingleNodeSingleSharedPoolSizeClusterConfig,
+    CitusSingleNodeSingleConnectionClusterConfig,
+    CitusSingleWorkerClusterConfig,
+    CitusShardReplicationFactorClusterConfig,
+    CitusNoLocalExecutionClusterConfig,
+    CitusComplexClusterConfig,
+    CitusSuperUserDefaultClusterConfig,
+    CITUS_CUSTOM_TEST_DIR,
+    CUSTOM_TEST_NAMES,
+    COORDINATOR_NAME,
+    CUSTOM_CREATE_SCHEDULE,
+    CUSTOM_SQL_SCHEDULE,
+    REGULAR_USER_NAME
+)
+
+testResults = {}
+failCount = 0
+
+def run_for_config(config, name):
+    global failCount
+    print ('Running test for: {}'.format(name))
+    start_time = time.time()
+    common.initialize_citus_cluster(config.bindir, config.datadir, config.settings, config)
+    if config.user == REGULAR_USER_NAME:
+        common.create_role(config.bindir, config.coordinator_port(),
+            config.node_name_to_ports.values(), config.user)
+    copy_test_files(config)
+
+    exitCode = common.run_pg_regress_without_exit(config.bindir, config.pg_srcdir,
+                   config.coordinator_port(), CUSTOM_CREATE_SCHEDULE, config.output_dir,
+                   config.input_dir, config.user)
+    common.save_regression_diff('create', config.output_dir)
+    if config.is_mx and config.worker_amount > 0:
+        exitCode |= common.run_pg_regress_without_exit(config.bindir, config.pg_srcdir,
+                   config.random_worker_port(), CUSTOM_SQL_SCHEDULE, config.output_dir, config.input_dir, config.user)
```

Comment:
It probably makes sense to run sql schedule on both coordinator and a random worker node but that would mean we will need to start the database from scratch because when rerunning "insert" statements, multiple runs will increase the table sizes hence test results will be different